### ISSUE TITLE
CLI/config option to disable parallel_compilation

### ISSUE THREAD

**Issue by TerrorJack** [bytecodealliance/wasmtime#4910]
Thanks for filing a feature request! Please fill out the TODOs below.

#### Feature

It would be nice to have a CLI/config option to disable `parallel_compilation`.

#### Benefit

As a part of an ongoing debugging effort, I'm repeatedly creating and executing many short-lived wasm modules in parallel on a many cores server. Having a way to disable `parallel_compilation` can greatly benefit this use case by avoiding over subscription of CPU.

#### Implementation

Doesn't sound too hard, if no objections I can give it a try.

#### Alternatives

Not really.
